### PR TITLE
[ILCompiler] Fix formatting error in EHClause.WriteTo

### PR DESCRIPTION
**Pull request by filipnavara**
Contributes to #96310

### CODE REVIEW

File: src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/EHInfo.cs [hunk 1]
```diff
@@ -132,11 +132,11 @@ public void WriteTo(TextWriter writer, int methodRva, bool dumpRva)
             switch (Flags & CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_KIND_MASK)
             {
                 case CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_NONE:
-                    writer.Write($" CATCH: {0}", ClassName ?? "null");
+                    writer.Write(" CATCH: {0}", ClassName ?? "null");
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/EHInfo.cs:135]:
```suggestion
                    writer.Write($" CATCH: {ClassName ?? "null"}");
```
The rest of the method uses interpolated strings. Keep using interpolated string here?

File: src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/EHInfo.cs [hunk 2]
```diff
@@ -132,11 +132,11 @@ public void WriteTo(TextWriter writer, int methodRva, bool dumpRva)
...
                     break;
 
                 case CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_FILTER:
-                    writer.Write($" FILTER (RVA {0:X4})", ClassTokenOrFilterOffset + methodRva);
+                    writer.Write(" FILTER (RVA {0:X4})", ClassTokenOrFilterOffset + methodRva);
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/EHInfo.cs:139]:
```suggestion
                    writer.Write(" FILTER (RVA {(ClassTokenOrFilterOffset + methodRva):X4})");
```

File: src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/EHInfo.cs [hunk 3]
```diff
@@ -132,11 +132,11 @@ public void WriteTo(TextWriter writer, int methodRva, bool dumpRva)
             switch (Flags & CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_KIND_MASK)
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/EHInfo.cs:132]: There are bugs in the interpolated strings above as well. For example,, `writer.Write(@"(RVA {(HandlerEnd + methodRva):X4}) ")` should be `writer.Write($"(RVA {(HandlerEnd + methodRva):X4}) ")`.

File: src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/EHInfo.cs (hunk 1 above)
Comment (filipnavara) [reply src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/EHInfo.cs:135]: I was torn which way is better; the rest of the code doesn't use complex expressions in the interpolations. I'm fine with either way.

File: src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/EHInfo.cs (hunk 3 above)
Comment (jkotas) [reply src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/EHInfo.cs:132]: Also, `$@` can be just `$`

Comment (filipnavara) [reply src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/EHInfo.cs:132]: Good catch, fixed.

**Review by jkotas — APPROVED** [review]
Thanks

**Review by stephentoub — APPROVED** [review]

**Pull request merged**